src/main/java/de/srendi/advancedperipherals/common/data/TurtleUpgradesProvider.java

```diff
@@ -1,37 +1,45 @@
 package de.srendi.advancedperipherals.common.data;
 
 import dan200.computercraft.api.turtle.ITurtleUpgrade;
-import dan200.computercraft.api.turtle.TurtleUpgradeDataProvider;
-import dan200.computercraft.api.upgrades.UpgradeSerialiser;
+import de.srendi.advancedperipherals.common.addons.computercraft.turtles.TurtleChatBoxUpgrade;
+import de.srendi.advancedperipherals.common.addons.computercraft.turtles.TurtleChunkyUpgrade;
+import de.srendi.advancedperipherals.common.addons.computercraft.turtles.TurtleCompassUpgrade;
+import de.srendi.advancedperipherals.common.addons.computercraft.turtles.TurtleEnvironmentDetectorUpgrade;
+import de.srendi.advancedperipherals.common.addons.computercraft.turtles.TurtleGeoScannerUpgrade;
+import de.srendi.advancedperipherals.common.addons.computercraft.turtles.TurtlePlayerDetectorUpgrade;
+import de.srendi.advancedperipherals.common.addons.computercraft.turtles.metaphysics.EndAutomata;
+import de.srendi.advancedperipherals.common.addons.computercraft.turtles.metaphysics.HusbandryAutomata;
+import de.srendi.advancedperipherals.common.addons.computercraft.turtles.metaphysics.OverpoweredEndAutomata;
+import de.srendi.advancedperipherals.common.addons.computercraft.turtles.metaphysics.OverpoweredHusbandryAutomata;
+import de.srendi.advancedperipherals.common.addons.computercraft.turtles.metaphysics.OverpoweredWeakAutomata;
+import de.srendi.advancedperipherals.common.addons.computercraft.turtles.metaphysics.WeakAutomata;
 import de.srendi.advancedperipherals.common.setup.Blocks;
 import de.srendi.advancedperipherals.common.setup.CCRegistration;
 import de.srendi.advancedperipherals.common.setup.Items;
-import net.minecraft.data.PackOutput;
-import org.jetbrains.annotations.NotNull;
+import net.minecraft.data.worldgen.BootstrapContext;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.item.ItemStack;
 
-import java.util.function.Consumer;
+public class TurtleUpgradesProvider {
 
-public class TurtleUpgradesProvider extends TurtleUpgradeDataProvider {
-
-    public TurtleUpgradesProvider(PackOutput generator) {
-        super(generator);
+    public static void addUpgrades(BootstrapContext<ITurtleUpgrade> upgrades) {
+        upgrades.register(id(CCRegistration.ID.CHATTY_TURTLE), new TurtleChatBoxUpgrade(new ItemStack(Blocks.CHAT_BOX.get())));
+        upgrades.register(id(CCRegistration.ID.PLAYER_TURTLE), new TurtlePlayerDetectorUpgrade(new ItemStack(Blocks.PLAYER_DETECTOR.get())));
+        upgrades.register(id(CCRegistration.ID.ENVIRONMENT_TURTLE), new TurtleEnvironmentDetectorUpgrade(new ItemStack(Blocks.ENVIRONMENT_DETECTOR.get())));
+        upgrades.register(id(CCRegistration.ID.CHUNKY_TURTLE), new TurtleChunkyUpgrade(new ItemStack(Items.CHUNK_CONTROLLER)));
```

Comment:
hm do we don't need Items.CHUNK_CONTROLLER`.get()` any more?

Reply:
Didn't catch that
But it seems that the ItemStack constructor only has a variant for `Holder<Item>` and not for `Holder<Block>`
So I can't remove the get from the other stacks